tests/src/Interop/IJW/NativeVarargs/IjwNativeVarargs.cpp

```diff
@@ -0,0 +1,474 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+#include <cstddef>
+#include <vector>
+#include <cstdarg>
+#include <numeric>
+#include <array>
+#include <functional>
+#include <iostream>
+#using <mscorlib.dll>
+using namespace System::Collections::Generic;
+
+public enum class TestCases
+{
+    SumInts,
+    SumFloats,
+    SumLongLongs,
+    SumPromotedShorts,
+    SumDoubles,
+    SumPromotedFloats,
+    SumHFAs
+};
+
+struct HFA
+{
+    float f1;
+    float f2;
+    float f3;
+    float f4;
+};
+
+#pragma unmanaged
+
+int SumInts(std::size_t numElements, ...)
+{
+    va_list args;
+    va_start(args, numElements);
+    int sum = 0;
+    for (std::size_t i = 0; i < numElements; ++i)
+    {
+        sum += va_arg(args, int);
+    }
+    va_end(args);
+    return sum;
+}
+
+long long SumLongLongs(std::size_t numElements, ...)
+{
+    va_list args;
+    va_start(args, numElements);
+    long long sum = 0;
+    for (std::size_t i = 0; i < numElements; ++i)
+    {
+        sum += va_arg(args, long long);
+    }
+    va_end(args);
+    return sum;
+}
+
+double SumDoubles(std::size_t numElements, ...)
+{
+    va_list args;
```

Comment:
This may be too much work; however, to test the x64 ABI it would be nice to **only** on x64 read doubles as __int64.